### DISCUSSION TITLE
Unchecked NullReference

### DISCUSSION THREAD

**Discussion by DeznekCZ** [dotnet/runtime#65269] [Ideas]
https://github.com/dotnet/runtime/blob/78c6505cffe2558b036fbe44cd27038affbb6cce/src/libraries/System.Linq/src/System/Linq/Utilities.cs#L71

Hi devs,
I discovered, that the linked line does not check, od result of first coverter is null.
Is there try catch on iterator or it stops execution?

**Comment by stephentoub** [comment]
what are you worried about being null?

**Reply by DeznekCZ** [reply]
In case the selector2 requires non null value.

**Reply by stephentoub** [reply]
That delegate is provided by the user. LINQ has no way of knowing whether null is a valid value or not. Any validation in that regard would need to be provided by the delegate itself, and any exception it throws would propagate out of the MoveNext call.

**Reply by DeznekCZ** [reply]
This 'it throws would propagate out of the MoveNext call' means that the element will be skipped or ends completely next execution.

**Reply by stephentoub** [reply]
May I suggest you try it and see?
